test/testdata/configs/preset_module.yml

```diff
@@ -0,0 +1,3 @@
+# TODO: host a preset in the golangci org?
+presets:
+ - github.com/thepwagner/golangci-lint-yaml
```

Comment:
Could it be located in `golangci-lint/test/testdata` ?

Reply:
Yep, but I'm not sold on that path: the goal of this test is to exercise loading the configuration from a go module.

Using a `replace` directive, we could load from a go module that happens to be housed at `./test/testdata/...`:
```
require github.com/golangci/golangci-lint-yaml v0.0.1
replace github.com/golangci/golangci-lint-yaml => ./test/testdata/some-new-dir
```

I figured since https://github.com/golangci/golangci-lint/issues/1141 mentions "presets like `golangci:recommended`", using such a preset here would be the best solution. To shoe-horn that idea into this proposal, a module like `github.com/golangci/golangci-recommended` would be the equivalent of `golangci:recommended` preset.

--- 

Any repo with a `.golangci.yml` should work - using an existing dependency like [github.com/ldez/gomoddirectives](https://github.com/ldez/gomoddirectives/blob/v0.2.2/.golangci.yml) would have been less egregious to the dependency graph, but I thought the pattern of a module containing _only_ the configuration was more interesting to include in the proposal.

Reply:
I see.

In my case I would like have any github or https path (not exactly the go module or some convention of names).

```yaml
presets:
  - local/path/config.yml
  - https://raw.githubusercontent.com/golangci/golangci-lint/master/.pre-commit-hooks.yaml
  - github:golangci/golangci-lint/master/subfolder/docs/common.yaml
```

Reply:
> any https path

Per https://github.com/golangci/golangci-lint/issues/2137#issuecomment-891621683 , I don't think `https://` paths would be a welcome addition. If you want an HTTPS resource, they can be `curl`-ed first and used as a file preset.

> any github path

That might be useful, e.g. to have multiple configurations in a single repository.

Would it be acceptable for that path to also be a valid go module? Again, I  wanted to avoid adding outgoing network calls, authentication etc. Only using files provided by `go mod` is an attempt to walk that line.

Reply:
It make sense. Thank you for the hint. 

In this case then I would propose first support only local path and think about go mod solution in separate PR.